### PR TITLE
[MM-20443/GH-16131] - Migrate 'components/integrations/installed_commands'  to TypeScript

### PR DESCRIPTION
**Pull request by lucievr**
#### Summary

- Migrated all files inside `components/integrations/installed_commands` to TypeScript
- Renamed files to have TypeScript extensions and updated incorrect imports
- Ran check-types which resulted in no errors

#### Ticket Link

Fixes [this GH issue](https://github.com/mattermost/mattermost-server/issues/16131)
https://mattermost.atlassian.net/browse/MM-20443

### CODE REVIEW

File: components/integrations/installed_commands/installed_commands.tsx [hunk 1]
```diff
@@ -1,71 +1,46 @@
...
-    }
+type Props = {
+    team: Team;
+    user: UserProfile;
+    users: RelationOneToOne<UserProfile, UserProfile>;
+    commands: Command[]; // Installed slash commands to display
+    loading: boolean;
+    canManageOthersSlashCommands: boolean; // Set to allow changes to installed slash commands
+    actions: {
+
```
Comment (nevyangelova) [inline components/integrations/installed_commands/installed_commands.tsx:25]: nit: empty line

File: components/integrations/installed_commands/installed_commands.tsx [hunk 2]
```diff
@@ -1,71 +1,46 @@
...
+        // The function to call when Regenerate Token link is clicked
+        regenCommandToken: (id: string) => Promise<ActionResult>;
+
```
Comment (nevyangelova) [inline components/integrations/installed_commands/installed_commands.tsx:28]: nit: empty line

File: components/integrations/installed_commands/installed_commands.tsx [hunk 3]
```diff
@@ -1,71 +1,46 @@
...
-            * The function to call when Delete link is clicked
-            */
-            deleteCommand: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    team: Team;
+    user: UserProfile;
+    users: RelationOneToOne<UserProfile, UserProfile>;
+    commands: Command[]; // Installed slash commands to display
```
Comment (nevyangelova) [inline components/integrations/installed_commands/installed_commands.tsx:21]: nit: please add comments above property

**Review by nevyangelova — APPROVED** [review]
Thanks! LGTM minus some nit-picky non-blocking comments 👍

**Comment by lucievr** [comment]
Thanks for the review, @nevyangelova! I have moved the comments to separate lines above each property as requested. As for the empty lines, from what I understand ESLint by default requires empty lines before line comments so I left it as is. Let me know if any more changes are needed.

**Review by hmhealey — APPROVED** [review]
Yeah, that's somewhere we require blank lines. There's some places that it does it that get really annoying, but I personally like it before comments.

Regardless, thanks for the PR!

**Comment by lucievr** [comment]
Thank you, @hmhealey!

**Review by jgilliam17 — APPROVED** [review]
Thank you for your contribution @lucievr 
Tested, looks good to merge.
- Verified list of custom slash commands on Integrations page is as before.

**Pull request merged**